src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -631,3 +681,1111 @@ def delete_project(self, project_id: UUIDLike) -> GlobusHTTPResponse:
+            When set to "private", only entities in the same project as the client can
+            view it.
+        :type visibility: str
+        :param redirect_uris: list of URIs that may be used in OAuth authorization
+            flows.
+        :type redirect_uris: iterable of str, optional
+        :param terms_and_conditions: URL of client's terms and conditions.
+        :type terms_and_conditions: str, optional
+        :param privacy_policy: URL of client's privacy policy.
+        :type privacy_policy: str, optional
+        :param required_idp: In order to use this client a user must have an identity
+            from this IdP in their identity set.
+        :type required_idp: str or uuid, optional
+        :param preselect_idp: This pre-selects the given IdP on the Globus Auth login
+            page if the user is not already authenticated.
+        :type preselect_idp: str or uuid, optional
+        :param additional_fields: Any additional parameters to be passed through.
+        :type additional_fields: dict, optional
+
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> project = ...
+                    >>> r = ac.create_client(
+                    ...     "My client",
+                    ...     True,
+                    ...     project,
+                    ...     True,
+                    ... )
+                    >>> client_id = r["client"]["id"]
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.create_client
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/clients``
+
+                .. extdoclink:: Create Client
+                    :ref: auth/reference/#create_client
+        """
+        body: dict[str, t.Any] = {
+            "name": name,
+            "public_client": public_client,
+            "project": str(project),
+            "visibility": visibility,
+            "redirect_uris": redirect_uris,
+        }
+
+        if required_idp is not utils.MISSING:
+            body["required_idp"] = str(required_idp)
+        if preselect_idp is not utils.MISSING:
+            body["preselect_idp"] = str(preselect_idp)
+
+        # terms_and_conditions and privacy_policy must both be set or unset
+        if type(terms_and_conditions) is not type(privacy_policy):
+            raise exc.GlobusSDKUsageError(
+                "terms_and_conditions and privacy_policy must both be set or unset"
+            )
+        links: dict[str, str] = {}
+        if not isinstance(terms_and_conditions, utils.MissingType):
+            links["terms_and_conditions"] = terms_and_conditions
+        if not isinstance(privacy_policy, utils.MissingType):
+            links["privacy_policy"] = privacy_policy
+
+        if links:
+            body["links"] = links
```

Comment:
Okay, this is another one for me to chew on.
As above, no need to change and :+1: for this way of doing it right now.

But maybe a good future would be... this?
```python
links = {
    "terms_and_conditions": terms_and_conditions,
    "privacy_policy": privacy_policy,
}
if utils.MissingType.any_values(links):
    body["links"] = links
```
or maybe even it's
```python
body["links"] = utils.MissingType.subdict(
    {
        "terms_and_conditions": terms_and_conditions,
        "privacy_policy": privacy_policy,
    }
)
```
which returns `MISSING` if no values are non-`MISSING`?

Lots to think about here; and, again, as someone who just implemented this stuff I'd love your feedback or ideas.

Reply:
It's a little goofy since you may come across APIs where you want the field to be included but empty: `{"links": {}}` and others where it shouldn't `{"links": MISSING}` (or others where it doesn't matter).